redash/handlers/metr_widgets.py

```diff
@@ -0,0 +1,34 @@
+
+from flask import request
+
+from redash import models
+from redash.handlers.base import BaseResource
+from redash.permissions import (
+    require_object_modify_permission,
+    require_permission
+)
+from redash.serializers import serialize_widget
+
+class MetrWidgetTagsResource(BaseResource):
+    """
+    Resource for handling tags of a metrWidget.
+
+    This resource provides methods for retrieving and updating tags associated with a metrWidget.
+    """
+
+    @require_permission("edit_dashboard")
+    def post(self, widget_id):
+        widget = models.Widget.get_by_id_and_org(widget_id, self.current_org)
+        require_object_modify_permission(widget.dashboard, self.current_user)
+        widget_properties = request.get_json(force=True)
+        tags = widget_properties["tags"]
+
+        metrWidget = models.metrWidget.query.filter_by(widget=widget).first()
+        if metrWidget:
+            metrWidget.tags = tags
+        else:
+            metrWidget = models.metrWidget(widget=widget, tags=tags)
+            models.db.session.add(metrWidget)
+
+        models.db.session.commit()
```

Comment:
I was checking yesterday if i do really need to add it again to the session and the documentation was saying that we only add it if new or not already in session, so i used `inspect` 
```
from sqlalchemy import inspect

state = inspect(metr_widget)
if state.persistent:
    print("The object is persistent and in the session.")
```
and it seemed to be already in the session but i was confused because i saw in other code in redash they also add it again to the session after an update.

So i believe that yes i will add it again just in case

Reply:
please find the update [here in this commit ](https://github.com/metr-systems/redash/pull/4/commits/7f337577ff4cca5ee75b2645d8884daf85ab7abd)

Reply:
It might as well be this and people just don't feel secure enough and add it again hehe